pages/Messages/Validation.md

```diff
@@ -0,0 +1,173 @@
+
+1. Content must be valid a valid JSON object as defined in [RFC7159](https://datatracker.ietf.org/doc/html/rfc7159).
+1. Content must include a type field matching one or more of the [core types](https://www.w3.org/TR/activitystreams-vocabulary/#h-types) or [extended types](https://www.w3.org/TR/activitystreams-vocabulary/#h-extendedtypes) defined in the [Activity Vocabulary](https://www.w3.org/TR/activitystreams-vocabulary/) specification.
+1. Content must include all necessary fields associated with the given type defined in the [Activity Pub](https://www.w3.org/TR/activitypub/) specification.
+
+Additional fields not required or defined by the Activity Pub specifications may also be included in accordance with various extensions of the specification, such as [Mastodon](https://docs.joinmastodon.org/spec/activitypub/), [ForgeFed](https://github.com/forgefed/forgefed) or one of the many [potential future extensions](https://www.w3.org/wiki/ActivityPub_extensions) proposed by the [W3C](https://www.w3.org).
+
+If the content of a message is no longer accessible, i.e. the URI of the message returns a 404 or 500 HTTP status, the message should no longer be considered valid.
+It is also recommended that implementations provide a warning either in the console or directly to the user with the associated HTTP status.
+For example, a message such as `"Content Inaccessible: Error 404"` would suffice.
+
+### Content Authenticity
+
+Authentication of a message's contents should be verified by hashing the exact contents of the body returned by the message URI and comparing it with the `contentHash` field of the given message.
+Given that the signature of the message is valid, this should serve as proof that the signing user posted the activity pub content.
+
+## Type Specific Validation
+
+Specific types of DSNP messages will also include their own specific validations depending on the semantic intent of the message type.
+These specific validations follow:
+
+### Broadcasts
+
+1. All broadcast messages with a `"type"` field of `"Follow"` in the activity pub content must include a date in the `"published"` field preceded by a follow [Graph Change](/Messages/Overview#GraphChange) DSNP message with no unfollow events between the time of the follow and the broadcast messages.
+1. All broadcast messages with a `"type"` field of `"Like"` or `"Dislike"` in their activity pub content must be preceded by a [Reaction](/Messages/Overview#Reaction) DSNP message referring to the same object, i.e. that the `URI` message field of DSNP message identified by the `inReplyTo` field of the reaction is identical to the `"object"` field of the like or dislike activity pub content.
+1. All broadcast messages with a `"type"` field of `"Undo"` must include an `"object"` field with either a valid DSNP identifier referring to a previously announced broadcast message or a URI matching the URI of a previously announced broadcast event.
```

Comment:
I wonder if we should have native support for this as well...

Reply:
Yeah, it's probably worth talking over in the next spec discussion meeting.